addons/safemode/XEH_postInit.sqf

```diff
@@ -27,3 +27,19 @@ if (!hasInterface) exitWith {};
     // Player HUD
     !_weaponSafe call FUNC(setSafeModeVisual);
 }] call CBA_fnc_addPlayerEventHandler;
+
+// Variables are transferred from corpse to new body and EH above triggers on respawn, which desyncs safeties
+// Therefore, clear variables and remove mouse button input blocking when controlled unit dies
+[QUOTE(ADDON), "Killed", {
```

Comment:
```suggestion
[QUOTE(ADDON), "Respawn", {
```
If respawn delay is exactly 0, then the new unit will copy variables over before the killed EH runs
this will swap the wiping to the newly respawned unit

Reply:
I would have thought it would run it before the variables were transferred, but it seems it's after.